\documentclass{article}
\usepackage{graphicx}
\usepackage[margin=2.5cm]{geometry}
\usepackage[labelfont=bf]{caption}
\usepackage{color}
\usepackage{xcite}
\usepackage{amsmath}
\usepackage{textcomp}

\renewcommand{\textfraction}{1.0}
\renewcommand{\floatpagefraction}{.9}
\newcommand\revised[1]{\textcolor{red}{#1}}
\renewcommand{\topfraction}{0.9}    % max fraction of floats at top
\renewcommand{\bottomfraction}{0.8} % max fraction of floats at bottom
\renewcommand{\textfraction}{0.07}  % allow minimal text w. figs

\makeatletter 
\renewcommand{\fnum@figure}{Supplementary \figurename~\thefigure}
\renewcommand{\fnum@table}{Supplementary \tablename~\thetable}
\makeatother

%\renewcommand{\thefigure}{\@arabic\c@figure} 
%\renewcommand{\thetable}{\@arabic\c@table} 

\externalcitedocument{prickle}

\usepackage{url}
\urlstyle{same}

\begin{document}

\begin{titlepage}
\vspace*{3cm}
\begin{center}

{\LARGE
Assessing the reliability of spike-in normalization for analyses of single-cell RNA sequencing data
\par}

\vspace{0.75cm}

{\Large 
    \textsc{Supplementary Materials}
\par
}
\vspace{0.75cm}

\large
by


\vspace{0.75cm}
Aaron T. L. Lun$^1$, Fernando J. Calero-Nieto$^2$, Liora Haim-Vilmovsky$^{3,4}$, \\
Bertie G\"ottgens$^2$, John C. Marioni$^{1,3,4}$

\vspace{1cm}
\begin{minipage}{0.9\textwidth}
\begin{flushleft} 
$^1$Cancer Research UK Cambridge Institute, University of Cambridge, Li Ka Shing Centre, Robinson Way, Cambridge CB2 0RE, United Kingdom \\[6pt]
$^2$Cambridge Institute for Medical Research,  University of Cambridge, Wellcome Trust/MRC Building, Hills Road, Cambridge CB2 0XY, United Kingdom \\[6pt]
$^3$EMBL European Bioinformatics Institute, Wellcome Genome Campus, Hinxton, Cambridge CB10 1SD, United Kingdom \\[6pt]
$^4$Wellcome Trust Sanger Institute, Wellcome Genome Campus, Hinxton, Cambridge CB10 1SA, United Kingdom \\[6pt]
\end{flushleft}
\end{minipage}

\vspace{1.5cm}
{\large \today{}}

\vspace*{\fill}
\end{center}
\end{titlepage}

\newcommand\variance{\mbox{var}}

\section{Further interpretation of the mathematical terms}
$R_{is}$ was introduced as the average capture efficiency in well $i$ for all transcripts in set $s$.
It ranges from 0 to 1 and scales $r_{t_s}$ to determine the actual capture rate for each $t_s$.
The most obvious interpretation of $R_{is}$ (and $r_{t_s}$) is that of the efficiency of reverse transcription, but it also describes the efficiency of PCR amplification and tagmentation in Smart-seq2.
Thus, no additional variables are necessary for the latter steps.

The term $l_s L_i$ describes the rate at which reads are obtained from cDNA fragments during high-throughput sequencing.
The $l_s$ constant represents the average sequencing efficiency for transcripts in $s$, as well as factors such as mappability that affect the final counts.
The interpretation of $L_i$ depends on whether library quantification was performed to equalize the amount of cDNA from each well prior to sequencing.
If not, $L_i$ will be constant across wells, with its exact value depending on the sequencing depth.
However, if quantification was performed, $L_i$ will theoretically depend on the other variables that contribute to $T_{is}$.
Specifically,
\[
    L_i = D_i \left[ \sum_{s} \left( l_s V_{is} R_{is} \sum_{t_s} r_{t_s} c_{t_s} \right) \right]^{-1}
\]
where $D_i$ is the total sequencing depth for well $i$ (in reads) and the outer sum is taken over the spike-in sets $s\in \{1, 2\}$ as well as the set of endogenous transcripts $s=g$.
In practice, $L_i$ is effectively independent of $V_{is}$ and $R_{is}$ for any particular spike-in set $s$.
This is because the denominator of the above expression is dominated by the vastly larger number of cDNA fragments from the set of endogenous transcripts.
Any correlation between $L_i$ and the other terms would be negligible compared to the biological variance of expression.

The error term $\varepsilon_{is}$ describes the variability due to sequencing noise in the spike-in counts for set $s$.
We defined its variance as $\sigma^2_{lib(s)}$, which implicitly assumes that there is no relationship between the variance and the mean of $T_{is}$.
This is a strong assumption given that mean-variance relationships are often observed in RNA-seq data \cite{mccarthy2012differential,law2014voom}.
However, we note that the distribution of $T_{is}$ is similar between separate-addition and premixed-addition wells within each plate (Supplementary Figure~\ref{fig:totals}).
If the mean of $T_{is}$ does not change, neither will the value of $\sigma^2_{lib(s)}$, regardless of the nature of the mean-variance relationship. 
This suggests that $\sigma^2_{lib(s)}$ will not change between $\variance(\theta_i)$ and $\variance(\theta^*_i)$, allowing calculation of $\sigma^2_{vol}$ from their difference.

\section{Implementation details for the downstream analyses}

\subsection{Data pre-processing}
Quality control was performed by removing libraries with outlier values for various metrics \cite{lun2016stepbystep}, including 
the log-transformed total read count across all genes and the log-transformed total number of expressed genes, where small outliers were removed; 
and the proportion of reads mapped to spike-in transcripts or mitochondrial genes, where large outliers were removed.
Outlier values were defined as those that were more than three median absolute deviations away from the median value in the specified direction.
Genes were also removed if the average count across all cells was below 1.
This filters out low-abundance genes that do not contain much information for stable inference.
For the data sets generated here, only the ERCC spike-in transcripts were used in the analyses below.
Counts for the SIRV transcripts were discarded for simplicity.

\subsection{Methods for detecting differentially expressed genes}
For DEG detection with edgeR v3.16.3, a NB GLM was fitted to the counts for each gene \cite{mccarthy2012differential} using a suitable design matrix.
The log-transformed total count for the spike-in transcripts was used as the offset for each library.
An abundance-dependent trend was fitted to the NB dispersions of all genes using the estimateDisp function.
Empirical Bayes shrinkage was performed towards this trend to obtain a shrunken NB dispersion for each gene.
The likelihood ratio test was applied to test for significant differences in expression between conditions for each gene. 
Finally, the Benjamini-Hochberg correction was applied to control the FDR.

For MAST v1.0.5, an effective library size was defined by multiplying the total spike-in count by a constant value $C$ for each library.
We set $C$ to the ratio of the average total count for the endogenous genes to the average total count for the spike-in transcripts, where each average was computed across all libraries.
This procedure simply adjusts the library size for each cell to reflect the difference in the spike-in totals across cells.
Counts were converted to count-per-million (CPM) values using the effective library sizes.
This is the same as spike-in normalization, as the sum of CPMs across spike-in transcripts is equal across libraries.
CPMs were log-transformed after adding a pseudo-count of 1.
For each gene, a hurdle model was fitted to the log-CPMs across all cells using the zlm.SingleCellAssay function with default parameters.
Along with the experimental factors, the proportion of genes with non-zero counts in each library was included as a covariate \cite{finak2015mast} in the model.
Putative DE genes between the relevant conditions were identified using the lrTest function.

\subsection{Methods for detecting highly variable genes}
The first HVG detection method was based on the approach described by Brennecke \textit{et al.} \cite{brennecke2013accounting}, with some modifications to the size factors in order to perform spike-in normalization.
Specifically, the spike-in totals were rescaled to have a mean of unity across all libraries, and the size factor for each library was defined as its rescaled spike-in total.
This represents spike-in normalization as each ``spike-in size factor'' scales the counts in each library so that the spike-in totals are the same. 
We did \textit{not} compute separate size factors for the endogenous genes and spike-in transcripts, as this would require the use of non-DE normalization methods.
The rest of the method was implemented as originally described, using the technicalCV2 function in the scran package with min.bio.disp set to zero.
If blocking factors were present, they were regressed out by log-transforming the counts; applying the removeBatchEffect function from the limma package \cite{ritchie2015limma} to the log-counts; and converting the corrected log-values back to the count scale, prior to using technicalCV2.

The second approach to detect HVGs was based on computing the variance of log-expression values \cite{lun2016stepbystep}.
For each count in each library, a normalized log-expression value was defined as the log-ratio of the count with the spike-in size factor for that library.
(A pseudo-count of 1 was added prior to log-transformation to avoid undefined values.)
The variance of log-expression was computed across all cells for each spike-in transcript.
A loess curve was fitted to the log-variance against the mean for all spike-in transcripts using the trendVar function in scran, representing the mean-variance relationship due to technical noise.
The biological component of the variance and a $p$-value was computed for each gene using the decomposeVar function.
If blocking factors were present, they were used to construct a design matrix for modelling in trendVar.

\subsection{Methods for dimensionality reduction and clustering}
Spike-in size factors were defined from the spike-in totals as previously described. 
HVG detection was performed using the variance-of-log-expression method, where HVGs were defined as genes detected at a FDR of 5\% and with biological components above 0.5.
PCA was performed on the normalized log-expression values of the HVGs, using the prcomp function from the stats package with scaling and centring.
The first two PCs were used as the coordinates for each cell in one PCA plot, while the first and third PCs were used as the coordinates in another plot.
Each point was coloured according to its annotated cell type \cite{segerstople2016single}.

The procedure above was repeated at each simulation iteration with new spike-in counts.
Coordinates of all cells in each simulated PCA plot were mapped onto the original plot, after scaling and rotating the coordaintes to eliminate differences between plots that were not relevant to interpretation.
Specifically:
\begin{itemize}
    \item For each cell, the simulated coordinates was right-multiplied by a 2-by-2 transformation matrix
        \[
            \left[\begin{array}{cc}
                    b_x \cos(\psi) & - b_y \sin(\psi) \\
                    b_x \sin(\psi) & b_y \cos(\psi) 
                \end{array}
            \right]
        \]
        where $\psi$ is the angle around the origin and $b_x$ and $b_y$ scale the $x$- and $y$-coordinates respectively.
    \item The squared Euclidean distance from the (scaled and rotated) simulated coordinates to the original coordinates was computed for each cell, and summed across all cells.
    \item The scaling and rotation parameters of the matrix were identified that minimized the sum of squared distances across all cells, using the optim function from the stats package.
\end{itemize}
The more obvious approach to remapping is to directly project the simulated log-expression data onto the space of the original plot.
However, we do not do this as it does not capture the variability in the identification of the PCs across iterations.
Upon completion of the simulation, each cell will have one original location and one remapped location per iteration. 
For each cell, the smallest circle centered at its original location was drawn that contained the 95\% of the remapped locations.
(This avoids inflated circles due to outliers.)

Note that, for the data set used in this simulation, we removed cells that were already annotated as low quality in the associated metadata.
No additional outlier-based quality control was performed.
Moreover, we only used the cells extracted from a single individual (HP1502401, healthy male) for simplicity.

\begin{figure}[btp]
    \begin{center}
        \includegraphics[width=0.49\textwidth,trim=0mm 5mm 0mm 5mm,clip]{../real/pics/qq_separate.pdf}
        \includegraphics[width=0.49\textwidth,trim=0mm 5mm 0mm 5mm,clip]{../real/pics/qq_premixed.pdf}
    \end{center}
    \caption{Quantile-quantile plots of the log-ratios after separate addition of spike-ins (left) or premixed addition (right).
        For each plate, a linear model was fitted to the log-ratios to account for the experimental design.
        Residuals were standardized and plotted against the theoretical quantiles of a standard normal distribution.
        The dotted line represents equality between the sample and theoretical quantiles.
    }
\end{figure}

\begin{figure}[btp]
    \begin{center}
        \includegraphics[width=0.49\textwidth,trim=0mm 5mm 0mm 5mm,clip]{../real/pics/total_ercc.pdf}
        \includegraphics[width=0.49\textwidth,trim=0mm 5mm 0mm 5mm,clip]{../real/pics/total_sirv.pdf}
    \end{center}
    \caption{Distribution of the total number of reads assigned to transcripts in the ERCC (left) or SIRV spike-in set (right) across wells.
        For each plate, separate boxplots are shown for wells in which spike-ins were added separately or premixed before addition.
        Dots represent wells with total counts that are more than 1.5 interquartile ranges from the first or third quartile of the corresponding distribution.
    }
    \label{fig:totals}
\end{figure}

\begin{figure}[btp]
    \begin{center}
        \includegraphics[width=0.7\textwidth,trim=0mm 10mm 0mm 10mm,clip]{../real/pics/variance_order.pdf}
    \end{center}
    \caption{Estimated variance of the log-ratio of total counts between spike-in sets, computed across wells in which the ERCC spike-in set was added before the SIRV set or vice versa.
        This exploits the presence of a number of wells in each plate for which the order of spike-in addition was reversed.
        Error bars represent standard errors of the variance estimates for normally distributed log-ratios.
        Differences between the ERCC- and SIRV-first variance estimates of each batch were assessed using a two-sided F-test, yielding $p$-values of 0.96, 0.66, 0.02 and 0.33 for the respective batches from left to right.
    }
\end{figure}

\begin{figure}[btp]
    \begin{center}
        \includegraphics[width=0.49\textwidth,trim=0mm 10mm 0mm 10mm,clip,page=1]{../sequence_check/biophysical/comparison.pdf}
        \includegraphics[width=0.49\textwidth,trim=0mm 10mm 0mm 10mm,clip,page=2]{../sequence_check/biophysical/comparison.pdf}
    \end{center}
    \caption{Biophysical properties of transcripts in each of the two spike-in sets and for 2000 randomly selected transcripts from the mouse mm10 genome.
    Boxplots are shown for the distribution of lengths and GC contents of transcripts (not including the poly-A tail) in each set.
}
\end{figure}

\begin{table}[btp]
    \caption{Alignment and counting statistics for each batch of scRNA-seq data, including the total number of fragments (reads for single-end data, read pairs for paired-end data), percentage of reads mapped to the reference genome and percentage of fragments assigned to genic regions.
    For each statistic, the median value across all wells in the batch is shown with first and third quartiles in brackets.}
    \begin{center}
        \begin{tabular}{l r r r}
            \hline
            \textbf{Data set} & \textbf{Total ($\times 10^6$)} & \textbf{Mapped (\%)} & \textbf{Counted (\%)} \\
            \hline
            416B (I)  & 2.80 (2.39-3.10) & 59.2 (56.6-61.6) & 46.3 (45.1-49.3) \\
            416B (II) & 2.82 (2.40-3.26) & 50.3 (47.3-53.1) & 39.0 (36.5-42.3) \\
            Tropho (I) & 2.02 (17.9-2.20) & 88.8 (88.1-89.3) & 74.9 (73.0-76.5) \\
            Tropho (II) & 2.33 (2.08-2.57) & 89.1 (87.6-89.7) & 62.8 (61.5-65.5) \\
            \hline
        \end{tabular}
    \end{center}
\end{table}

% blah <- read.table("my_qual.tsv", header=TRUE, comment="")
% totals <- rowSums(blah)
% mapped <- 1 - blah$Unassigned_Unmapped/totals
% counted <- blah$Assigned/totals
% summary(totals, digits=100)
% summary(mapped)
% summary(counted)

\end{document}


